tests/ui/issues/issue-100605.stderr

```diff
@@ -27,10 +23,8 @@ error[E0308]: mismatched types
   --> $DIR/issue-100605.rs:8:18
    |
 LL |     takes_option(&res);
-   |     ------------ ^^^^
-   |     |            |
-   |     |            expected `Option<&String>`, found `&Option<String>`
-   |     |            help: you can convert from `&Option<T>` to `Option<&T>` using `.as_ref()`: `res.as_ref()`
+   |     ------------ ^^^^ expected `Option<&String>`, found `&Option<String>`
+   |     |
```

Comment:
Sad to lose these.

Reply:
I'll see if I can add it back

Reply:
Wait, I'm not actually sure what you're talking about when looking at this now 😅 

For the `Some` case, we still give the right suggestion below:

```
help: try using `.as_ref()` to convert `&Option<String>` to `Option<&String>`
   |
LL -     takes_option(&res);
LL +     takes_option(res.as_ref());
```

And for the `None` case, we no longer suggest `None.as_ref()`, in favor of just suggesting removing the `&` which is far more accurate.